iotdb-core/metrics/interface/src/main/java/org/apache/iotdb/metrics/metricsets/system/SystemMetrics.java

```diff
@@ -166,6 +173,48 @@ private void collectSystemMemInfo(AbstractMetricService metricService) {
         a -> osMxBean.getCommittedVirtualMemorySize(),
         SystemTag.NAME.toString(),
         SYSTEM);
+    if (CONFIG.getSystemType() == SystemType.LINUX) {
+      metricService.createAutoGauge(
+          SystemMetric.LINUX_MEMORY_COUNT.toString(),
+          MetricLevel.CORE,
+          this,
+          a -> a.updateLinuxSystemMemInfo(metricService),
+          SystemTag.NAME.toString());
+    }
+  }
+
+  private final String[] linuxMemoryTitles =
+      new String[] {"Total", "Used", "Free", "Shared", "Buff/Cache", "Available"};
+
+  private long updateLinuxSystemMemInfo(AbstractMetricService metricService) {
+    long count = 0;
+    try {
+      Process process = runtime.exec(getSystemMemoryCommand);
+      StringBuilder result = new StringBuilder();
+      try (BufferedReader input =
+          new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+        String line;
+        while ((line = input.readLine()) != null) {
+          result.append(line).append("\n");
+        }
+      }
+      logger.error(result.toString());
+      String[] lines = result.toString().split("\n");
+      String[] memParts = lines[1].trim().split("\\s+");
+      count = memParts.length;
+      for (int i = 1; i < count; i++) {
+        metricService
+            .getOrCreateGauge(
+                SystemMetric.LINUX_MEMORY_SIZE.toString(),
+                MetricLevel.CORE,
+                SystemTag.NAME.toString(),
+                linuxMemoryTitles[i - 1])
+            .set(Long.parseLong(memParts[i]) * 1024);
```

Comment:
do we need to multiple 1024? or we just need to set it in dashboard?

Reply:
I think it's a good idea to keep the union of related metrics same